File: services/activity-service/app/models/meeting_request.py
# meeting_request.py

import uuid
from datetime import datetime
from sqlalchemy import Column, String, DateTime, ForeignKey
from sqlalchemy.dialects.postgresql import UUID
from app.db.base import Base

class MeetingRequest(Base):
    __tablename__ = "meeting_requests"

    id = Column(String, primary_key=True, default=lambda: str(uuid.uuid4()))
    activity_id = Column(UUID(as_uuid=True), ForeignKey("activities.id"), nullable=False)
    user_id = Column(String, nullable=False)
    status = Column(String, default="pending")  # pending, approved, rejected
    created_at = Column(DateTime, default=datetime.utcnow)

File: mobile/lib/meetings/meeting_request.dart
import '../user/user.dart';

class MeetingRequest {
  final String id;
  final String activityId;
  final String userId;
  final String status; // pending, approved, rejected
  final DateTime createdAt;

  const MeetingRequest({
    required this.id,
    required this.activityId,
    required this.userId,
    required this.status,
    required this.createdAt,
  });

  factory MeetingRequest.fromJson(Map<String, dynamic> json) => MeetingRequest(
    id: json['id'] as String,
    activityId: json['activity_id'] as String,
    userId: json['user_id'] as String,
    status: json['status'] as String,
    createdAt: DateTime.parse(json['created_at'] as String),
  );
}

class MeetingRequestWithUser {
  final MeetingRequest request;
  final UserProfile? user;

  const MeetingRequestWithUser({required this.request, this.user});
}
